<a href="https://colab.research.google.com/github/ishanku/30-seconds-of-code/blob/master/Store.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [29]:
import numpy as np
import pandas as pd
import tensorflow as tf
from matplotlib import pyplot as plt
import statsmodels.formula.api as sm

In [27]:
!ls drive/MyDrive/store.csv

drive/MyDrive/store.csv


In [28]:
df = pd.read_csv('drive/MyDrive/store.csv')

In [33]:
model = sm.ols('Sales ~ MTenure + CTenure + ServQual', data=df)
results = model.fit()

In [32]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.321
Model:                            OLS   Adj. R-squared:                  0.292
Method:                 Least Squares   F-statistic:                     11.18
Date:                Sat, 11 Jan 2025   Prob (F-statistic):           4.28e-06
Time:                        15:42:22   Log-Likelihood:                -1038.4
No. Observations:                  75   AIC:                             2085.
Df Residuals:                      71   BIC:                             2094.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   4.349e+05   2.09e+05      2.084      0.0

##MTenure:
##Estimate: 1925.1695
##P-value: 0.001

###Every one-unit increase in 'MTenure', 'Sales' is expected to increase by 1925.1695 units, holding other variables constant.

### CTenure:
#### Estimate: 2418.1040
#### P-value: 0.168
##### This suggests that for every one-unit increase in 'CTenure', 'Sales' is expected to increase by 2418.1040 units, holding other variables constant. However, the p-value is greater than 0.05, indicating that this coefficient is not statistically significant. This means that we don't have enough evidence to conclude that 'CTenure' has a significant impact on 'Sales'.

### ServQual:
#### Estimate: 7453.9557
#### P-value: 0.003
##### This implies that for every one-unit increase in 'ServQual', 'Sales' is expected to increase by 7453.9557 units, holding other variables constant. The p-value is less than 0.05, indicating that this coefficient is statistically significant.
###### 1.'MTenure' and 'ServQual' have statistically significant positive effects on 'Sales'.
###### 2.'CTenure' does not have a statistically significant effect on 'Sales' based on this model.

In [36]:
model = sm.ols('Sales ~ MTenure + CTenure + + PedCount +  ServQual + Pop + Comp + MgrSkill + CrewSkill', data=df)
results = model.fit()

In [38]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.581
Model:                            OLS   Adj. R-squared:                  0.530
Method:                 Least Squares   F-statistic:                     11.44
Date:                Sat, 11 Jan 2025   Prob (F-statistic):           5.09e-10
Time:                        16:02:33   Log-Likelihood:                -1020.3
No. Observations:                  75   AIC:                             2059.
Df Residuals:                      66   BIC:                             2079.
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1.171e+05   3.43e+05      0.341      0.7


R-squared: 0.581 - This indicates that the model explains 58.1% of the variance in 'Sales'. This is a moderate level of explanatory power.

Adj. R-squared: 0.530 - This is a slightly lower value than R-squared, adjusted for the number of predictors in the model. It's still a decent fit.

F-statistic: 11.44, with a very low p-value (5.09e-10) - This indicates that
the overall model is statistically significant, meaning that at least one of the predictors has a significant relationship with 'Sales'.
```
# This is formatted as code
```



Coefficients and their Significance:

Intercept: 1.171e+05 (117,100) with a p-value of 0.734 - This is not
statistically significant, meaning it's not reliably different from zero.

MTenure: 2038.2223 with a p-value of 0.000 - This is statistically significant and positive. It suggests that for every one-unit increase in 'MTenure', 'Sales' is expected to increase by about 2038 units, holding other variables
constant.

CTenure: 2265.5458 with a p-value of 0.136 - This is not statistically significant at the conventional 0.05 level. While the coefficient is positive, there's not enough evidence to say it has a reliable impact on 'Sales'.

PedCount: 8.297e+04 (82,970) with a p-value of 0.011 - This is statistically significant and positive. An increase in 'PedCount' is associated with a substantial increase in 'Sales'.

ServQual: 5843.2207 with a p-value of 0.007 - This is statistically significant and positive, indicating a positive relationship with 'Sales'.

Pop: 8.1030 with a p-value of 0.151 - This is not statistically significant.

Comp: -5.803e+04 (-58,030) with a p-value of 0.006 - This is statistically significant and negative, suggesting that an increase in 'Comp' is associated with a decrease in 'Sales'.

MgrSkill: 1.039e+05 (103,900) with a p-value of 0.128 - This is not statistically significant.

CrewSkill: -8128.1265 with a p-value of 0.902 - This is not statistically significant and has a negative coefficient, but it's not reliable given the high p-value.

In summary:
The model suggests that 'MTenure', 'PedCount', 'ServQual', and 'Comp' are the most important predictors of 'Sales' in this dataset. 'MTenure', 'PedCount', and 'ServQual' have positive relationships with 'Sales', while 'Comp' has a negative relationship. The other variables in the model do not have statistically significant effects on 'Sales' based on these results.
Important Notes:
Causation: Remember that correlation does not equal causation. These results show associations, but further analysis would be needed to establish causal relationships.

Assumptions: OLS regression has assumptions (linearity, independence of errors, etc.). Does these assumptions are met for the validity of the results ?

Context: Interpretation of the results in the context of specific business or research question can impact. The practical significance of the coefficients is important to consider alongside statistical significance.